src/ceph-volume/ceph_volume/util/encryption.py

```diff
@@ -106,17 +106,26 @@ def luks_open(key, device, mapping):
     process.call(command, stdin=key, terminal_verbose=True, show_command=True)
 
 
-def dmcrypt_close(mapping):
+def dmcrypt_close(mapping_path):
     """
     Encrypt (close) a device, previously decrypted with cryptsetup
 
-    :param mapping:
+    :param mapping_path:
     """
-    if not os.path.exists(mapping):
+    if not os.path.exists(mapping_path):
         logger.debug('device mapper path does not exist %s' % mapping)
         logger.debug('will skip cryptsetup removal')
         return
     # don't be strict about the remove call, but still warn on the terminal if it fails
+    process.run(['cryptsetup', 'remove', mapping_path], stop_on_error=False)
+
+def dmcrypt_close_by_name(mapping):
+    """
+    Encrypt (close) a device, previously decrypted with cryptsetup
+
+    :param mapping: dmcrypt device name
+    """
+    # don't be strict about the remove call, but still warn on the terminal if it fails
```

Comment:
What is the difference between `dmcrypt_close()` and `dmcrypt_close_by_name()` ?

At the end, they both do the same call to `process.run()`

Reply:
If this is a question passing a mapping name vs. a path, perhaps `dmcrypt_close()` should probably just be reworked a bit.
What do you think?

Reply:
I can see two points:
1. dmcrypt_close vs. dmcrypt_close_by_name operate on different input parameters: path vs. name. Hence better to distinguish that somehow
2. dmcrypt_close performs additional check on path presence. Additional checks (processing logic) might appear in the future at one of implementations - thus again better to have them different

I'm not very good at Python-style coding so please advice if you think there are more 'pythonish" approaches to implement the above.

Reply:
I would probably just adapt the existing function like this instead:


```python
def dmcrypt_close(mapping, skip_path_check=False):
    """
    Encrypt (close) a device, previously decrypted with cryptsetup

    :param mapping:
    """
    if not skip_path_check and not os.path.exists(mapping):
        logger.debug('device mapper path does not exist %s' % mapping)
        logger.debug('will skip cryptsetup removal')
        return
    # don't be strict about the remove call, but still warn on the terminal if it fails
    process.run(['cryptsetup', 'remove', mapping], stop_on_error=stop_on_error)
```
it renames `mapping_path` to `mapping` which, IMO, becomes a bit more generic and adds an argument so you can decide to skip the mapper path check.
(note that it fixes a bug in the line `logger.debug('device mapper path does not exist %s' % mapping)`, `mapping` was undefined so far...)

Reply:
@guits - updated as per you suggestion.